### DISCUSSION TITLE
Actions: Third-party authorization

### DISCUSSION THREAD

**Discussion by geoffatsource** [hasura/graphql-engine#9392] [Question] [answered]
I am trying to pass-through a bearer token to authenticate with a third-party API so I am putting into the headers the following:

Action Definition:
```
type Query {
  """ Read a list of quickbooks vendors """ qb_query_vendors(
    access_token: String
  ): [Vendor]
}
```

Headers
```
Key: Authorization
Value: {{$body.input.access_token}}
```

but it appears to be sending the literal value `{{$body.input.access_token}}` and not parsing as a template

How can I use an input and pass it through as a header value? 
Or, how can I pass a hasura request header through with a different header key?

**Comment by ajohnson1200** [comment, marked as answer]
Filed as an internal support discussion as well, for posterity / future readers:

> Currently, we do not parse header literal values even if they are templated values .
 > You can try modifying your action to one of these approaches
Choose select "From env var" value from dropdown while defining headers in create action form and pass your third party token env var to it.
> Choose "Forward client headers to webhook" and then pass dynamic value of token from your client in form of headers while querying against Hasura
> Consume your token from action input payload in your backend middleware and send it as headers forward to your business logic
